flow/layers/layer_tree.cc

```diff
@@ -19,25 +19,33 @@ LayerTree::LayerTree()
 
 LayerTree::~LayerTree() = default;
 
-void LayerTree::Preroll(CompositorContext::ScopedFrame& frame,
-                        bool ignore_raster_cache) {
+// Executes preroll on the root layer of the tree.
+//
+// Returns size hints for the surfaces needed to draw the tree. This may
+// be used to make more intelligent decisions such as when on fuchsia, we
+// do smart allocation of Vulkan surfaces in the VulkanSurfacePool.
+std::vector<SkISize> LayerTree::Preroll(CompositorContext::ScopedFrame& frame,
+                                        bool ignore_raster_cache) {
   TRACE_EVENT0("flutter", "LayerTree::Preroll");
   SkColorSpace* color_space =
       frame.canvas() ? frame.canvas()->imageInfo().colorSpace() : nullptr;
   frame.context().raster_cache().SetCheckboardCacheImages(
       checkerboard_raster_cache_images_);
+  std::vector<SkISize> sizes;
   PrerollContext context = {
       ignore_raster_cache ? nullptr : &frame.context().raster_cache(),
       frame.gr_context(),
       frame.view_embedder(),
       color_space,
       SkRect::MakeEmpty(),
+      &sizes,
       frame.context().frame_time(),
       frame.context().engine_time(),
       frame.context().texture_registry(),
       checkerboard_offscreen_layers_};
 
   root_layer_->Preroll(&context, frame.root_surface_transformation());
+  return sizes;
```

Comment:
Can we have a link to the source/doc that actually uses sizes? From the change in this PR alone, it's not clear to me how they're going to be used.

Reply:
The code that will be using sizes is Fuchsia, and as far as I know has not been written/completed yet.